mkinitcpio

```diff
@@ -251,6 +260,56 @@ build_image() {
     fi
 }
 
+build_uefi(){
+    set -x
+    local out=$1 microcode=$2 initramfs=$3 cmdline=$4 osrelease=$5 splash=$6 kernelimg=$7 uefistub=$8 errmsg=
+    OBJCOPYARGS=()
+
+    if [[ -z "$uefistub" ]]; then
+        for stub in {/usr,}/lib/systemd/boot/efi/linux{x64,ia32}.efi.stub {/usr,}/lib/gummiboot/linux{x64,ia32}.efi.stub; do
+            if [[ -f "$stub" ]]; then
+                uefistub="$stub"
+                break
+            fi
+        done
+    fi
+
+    if [[ -z "$kernelimg" ]]; then
+        for img in "/lib/modules/$KERNELIMAGE" "/boot/vmlinuz-$KERNELIMAGE" "/boot/vmlinuz-linuz"; do
```

Comment:
This line is broken! Missing lookup with $KERNELVERSION D: and "vmlinuz-linuz"

Reply:
Some tools do backups of `/lib/modules/$KERNELIMAGE` in roder to not lose modules of running kernel after live update so maybe it's better to skip that path and use only `/boot`

Reply:
> Some tools do backups of `/lib/modules/$KERNELIMAGE` in order to not lose modules of running kernel after live update so maybe it's better to skip that path and use only `/boot`

That shouldn't be a problem; mkinitcpio already knows which kernel version it's being invoked for (I mean, a _large_ part of its job is to copy the correct modules from /lib/modules) so it'll just take the kernel image from the same directory, I assume.

Reply:
I don't consider that a problem. We should know which kernel we are building the initramfs for. If someone points the wrong kernel at it without any modules in `/lib/modules` then it's a user error. We issue an error then anyway.

Reply:
I meant the opposite situation - the modules exist but kernel at `/boot` doesn't however if the version is known beforehand then it should be ok, thx.

Reply:
If the modules directory exists and `/boot` doesn't have a kernel, then you should still have `/lib/modules/$KERNELVERSION/vmlinuz`?

Reply:
Yes but building image out of it is never good idea as it may lead to issues like [this](https://github.com/andreyv/sbupdate/issues/38) one. This is why I wanted to confirm mkinitcpio won't be affected by same problem.

Reply:
That issue is rooted in `sbupdate` being unaware of the correct kernel version and guessing from the modules directory. That is a mistake which can't happen in `mkinitcpio`.